### PR TITLE
Final Amendments To Use Case Section

### PR DESCRIPTION
**Pull request by HM33-Stu**
Final Amendments To Use Case Section

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -486,7 +486,7 @@ When adding a new internship, optional fields such as location and role may rema
 **Editing Existing Internship:** <br>
 Users can edit existing internship details, including optional fields, to update or modify internship information as needed.
 
-#### Alternative Approach and Cons
+#### Design Considerations
```
Comment (HM33-Stu) [inline docs/DeveloperGuide.md:489]: @LWS49 Change made!

**Comment by HM33-Stu** [comment]
This is the main change for all use cases:
image
- To follow the Use Cases (particularly MSS and Extensions) format from Tutorial
- No "Use case ends." at the end of extensions. It needs to be inside each extension.
- If the extension is an error, then "Use case ends."
- If the extension is just some invalid value, then ""Use case resumes from Step X."

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -612,77 +616,78 @@ Use case ends.
...
+  * 3a1. System displays a message indicating no matching results found. <br>
+    Use case ends.
 
 **Use Case: Add Internship Task From Main Window**
 
-**MSS**
+**MSS:**
 1. User accesses the main window.
 2. User uses the command to specify which internship entry to add a task to and the task details.
 3. System validates that the task is valid.
```
[the remark this answers is not in the document]
Comment (jieqiboh) [reply docs/DeveloperGuide.md:693]: Change extensions accordingly

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -691,65 +696,64 @@ Use case ends.
...
+* 2a. System recognizes that user has entered invalid task details. <br>
+  * 2a1. System displays an error message indicating the task detail error(s). <br>
+    Use case resumes from Step 2.
 
 **Use Case: Delete Internship Task From Main Window**
 
-**MSS**
+**MSS:**
 1. User accesses the main window. 
 2. User uses the command to specify which internship task to delete.
```
Comment (jieqiboh) [inline docs/DeveloperGuide.md:712]: Specifying internship entry and task to delete from it
This appears for a few other MSS as well

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -691,65 +696,64 @@ Use case ends.
...
+  * 2a1. System displays an error message indicating the index error. <br>
+    Use case resumes from Step 2.
 
 **Use Case: Set Deadline For Internship Task From Main Window**
 
-**MSS**
+**MSS:**
 1. User accesses the main window.
 2. User uses the command to specify which internship task to set a deadline for. 
 3. System displays full list of internships.
```
Comment (jieqiboh) [inline docs/DeveloperGuide.md:731]: No validation or deadline set step

**Pull request merged**